overlord/devicestate/devicemgr.go

```diff
@@ -329,22 +335,74 @@ func (m *DeviceManager) maybeSetupUbuntuSave() error {
 	if !runMntSaveMounted {
 		// we don't have ubuntu-save, save will be used directly
 		logger.Noticef("no ubuntu-save mount")
-		m.saveAvailable = true
 		return nil
 	}
 
-	logger.Noticef("bind-mounting ubuntu-save under %v", dirs.SnapSaveDir)
+	// In newer core20/core22 we have a mount unit for ubuntu-save, which we
+	// we will try to start first. Invoking systemd-mount in this case would fail.
+	err = exec.Command("systemctl", "enable", "var-lib-snapd-save.mount", "--now").Run()
+	if err == nil {
+		logger.Noticef("mount unit for ubuntu-save was started")
+		return nil
+	}
 
```

Comment:
we tend to use the systemd package and not call systemd directly from inside snapd

Reply:
Fixed!